docs/source/interaction_model.rst

```diff
@@ -1,95 +1,75 @@
-+-----------+----------------------------------------------------------+
-| RECORD    | Record the rated items                                   |
-+-----------+----------------------------------------------------------+
+An example of interaction model is available at: `data/interaction_models/crs_v1.yaml`.
+
+New interaction models can be added by providing a YAML file with the same format as the example above. The path to this file can be provided either in the configuration file or the command line, see :ref:`Configuration`.
+
+Agenda update
+-------------
+
+The agenda is updated based on the last agent dialogue acts and the current state of the conversation. For each agent dialogue act, the interaction model performs a push operation on the agenda stack. We consider four cases:
+
+1. **Agent elicits**: In case the agent elicits information from the user, the interaction model may push a new dialogue act to disclose the information elicited.
+2. **Agent recommends**: In case the agent recommends an item, the interaction model may push a new dialogue act to express a preference regarding the recommended item (e.g., like, dislike, already consumed).
+3. **Agent inquires**: In case the agent inquires if the user wants to know more about a specific item, the interaction model may push a new dialogue act to request a slot in the information need or a random one.
+4. **None of the above**: In case none of the above cases apply, the interaction model checks if it is coherent to continue with the current agenda or if a new dialogue act should be sampled to keep the conversation going. In the latter case, the new dialogue act is sampled based on the transition probabilities from historical dialogues.
+
+Once all the push operations are done, the agenda is cleaned to discard unnecessary dialogue acts, e.g., duplicates.
+
+Transition probabilities matrices
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The interaction model uses transition probabilities matrices to sample new dialogue acts. These matrices are built from historical dialogues when the model is initialized. The transition probabilities are calculated based on the frequency of transitions between intents in the historical dialogues. We consider two matrices:
+
+* *Single intent*: The sequence of intents from an utterance is considered individually. That is, the transition probabilities are calculated based on the frequency of each intent following another intent.
+* *Compound intent*: The sequence of intents from an utterance is considered as a whole, i.e., the sequence of intents is considered as a single entity. That is, the transition probabilities are calculated based on the frequency of each sequence of intents following another sequence of intents.
+
+For example the following consecutive sequence of dialogue acts:
+
+| > Agent: [GREETINGS(), ELICIT(genre=?)]  
+| > User: [GREETINGS(), DISCLOSE(genre=action)]
+
+will result in the following transition probabilities matrices:
```

Comment:
Consider transposing the matrices, so that for each row (agent dialogue act) the columns would correspond to the user dialogue acts that might follow. To me this would be more natural if we're simulating users based on the last agent dialogue acts, but I could be wrong here.

Reply:
Fair point, it is easier to search by key which would corresponds to agent intent.